### ISSUE TITLE
How to document class/package options using `documentation` lib?

### ISSUE THREAD

**Issue by muzimuzhi** [T-F-S/tcolorbox#260]
It seems this issue was cleverly sidestepped in `tcolorbox` manual, probably because each of its package options has a corresponding key `/tcb/library/<pkg option>` defined which can be used as `\tcbuselibrary{<pkg option>, <pkg option>, ...}`.

`csvsimple.pdf` documents its two package options alongside with the introduction to `csvsimple-l3` and `csvsimple-lagency` packages.

I also found that the names `\colOpt{<text>}`, `/tcb/color option=<color>`, and color `Option` which may be considered suitable for options, are already taken by optional texts.

**Comment by T-F-S** [comment]
Is your own proposal https://github.com/T-F-S/tcolorbox/issues/259 a possible answer? With removing the root `\`, package options could be documented just like keys without path. Even your `/tcb/doclang/key` trick may be used to add something like *package option*, if that may be required.

**Comment by muzimuzhi** [comment]
> Is your own proposal #259 a possible answer?

Sure it is. But as I said in #259, from the partial key name `doclang`, I felt setting dynamic values to `/tcb/docloang/xxx` keys is more like a misuse.

**Comment by T-F-S** [comment]
> Sure it is. But as I said in #259, from the partial key name `doclang`, I felt setting dynamic values to `/tcb/docloang/xxx` keys is more like a misuse.

Yes, the naming is not inviting for the purpose. When I'm going to see for https://github.com/T-F-S/tcolorbox/issues/259, I try to keep that in mind and provide a solution.

**Comment by muzimuzhi** [comment]
Another aspect. `documentation` library provides two kinds of interfaces:
- long version which provides envs and an aux command, for example `docCommand(*)`, `docCommands`, and `\docAuxCommand` for commands
- short version which provides only a single command, like `\docLength` and `\docColor`.

If the decision is to provide a short version `\docOption` for package options, then my proposal(s) in #259 won't help.

**Comment by muzimuzhi** [comment]
Just one limitation: if `index colorize` is set to `true`, then indices for options and keys will always in the same color, the color (expression) passed to `/tcb/doc key` then stored in `\kvtcb@col@key`.

Writing value of `\kvtcb@col@key` to `.idx` will introduce limitation on the color (expression) stored in `\kvtcb@col@key`, as characters commonly used in color expressions like `!` have special meaning in `\index{<index entry>}`.

**Comment by muzimuzhi** [comment]
Typo: `/tcb/doc key` mentioned in my previous comment should all be `/tcb/color key`.

**Comment by T-F-S** [comment]
From my point of view, this is sufficiently solved by https://github.com/T-F-S/tcolorbox/issues/259
